test/harry/main/org/apache/cassandra/harry/gen/ValueGenerators.java

```diff
@@ -21,23 +21,29 @@
 import java.util.Comparator;
 import java.util.List;
 
-public class ValueGenerators
+import org.apache.cassandra.harry.gen.Bijections.Bijection;
+
+public class ValueGenerators<PartitionKey, ClusteringKey>
 {
-    protected final Bijections.Bijection<Object[]> pkGen;
-    protected final Bijections.Bijection<Object[]> ckGen;
+    protected final Bijection<PartitionKey> pkGen;
+    protected final Bijection<ClusteringKey> ckGen;
+
+    protected final Accessor<ClusteringKey> ckAccessor;
 
-    protected final List<Bijections.Bijection<Object>> regularColumnGens;
-    protected final List<Bijections.Bijection<Object>> staticColumnGens;
+    protected final List<? extends Bijection<? extends Object>> regularColumnGens;
+    protected final List<? extends Bijection<? extends Object>> staticColumnGens;
 
     protected final List<Comparator<Object>> pkComparators;
     protected final List<Comparator<Object>> ckComparators;
     protected final List<Comparator<Object>> regularComparators;
     protected final List<Comparator<Object>> staticComparators;
 
-    public ValueGenerators(Bijections.Bijection<Object[]> pkGen,
-                           Bijections.Bijection<Object[]> ckGen,
-                           List<Bijections.Bijection<Object>> regularColumnGens,
-                           List<Bijections.Bijection<Object>> staticColumnGens,
+    public ValueGenerators(Bijection<PartitionKey> pkGen,
+                           Bijection<ClusteringKey> ckGen,
+                           Accessor<ClusteringKey> ckAccessor,
+
+                           List<? extends Bijection<? extends Object>> regularColumnGens,
+                           List<? extends Bijection<? extends Object>> staticColumnGens,
```

Comment:
this is needed in order to allow the bijection cache to exist... working with `Object` is hard =D